ObjectTracking/src/bin/basic_control_1d.rs

```diff
@@ -172,8 +174,142 @@ impl<T: ToInputArray> ToInputArrayExt for T {
     }
 }
 
+// FIXME: this appears to use the debug representation if printed
+// to the console. We should make this prettier somehow.
+#[derive(Valuable)]
+struct JpegImage(Vec<u8>);
+
+impl JpegImage {
+    fn from_mat(mat: &Mat) -> Result<Self, eyre::Error> {
+        let mut buf = Default::default();
+        opencv::imgcodecs::imencode(".jpg", mat, &mut buf, &Default::default())?;
+        Ok(JpegImage(buf.to_vec()))
+    }
+}
+
+struct MJpegTracingSubscriber {
+    // mjpeg_rs::MJpeg doesn't *need* to be wrapped in a mutex,
+    // but it does not expose a way to tell whether its SyncSender
+    // would block, so let's hack around it, and submit a patch upstream later.
+    //
+    // We also need a Mutex to turn &self into &mut MJpegVisitor, so maybe it
+    // can stay like this for a while?
+    visitor: std::sync::Mutex<MJpegVisitor>,
+}
+
+impl MJpegTracingSubscriber {
+    fn new() -> Self {
+        Self {
+            visitor: std::sync::Mutex::new(MJpegVisitor::new()),
+        }
+    }
+    fn init(self) {
+        let mjpeg_server = self.visitor.lock().unwrap().mjpeg_server.clone();
+        std::thread::spawn(move || mjpeg_server.run("0.0.0.0:8088").unwrap());
+        tracing::dispatcher::set_global_default(self.into()).unwrap();
+    }
+}
+
+impl tracing::Subscriber for MJpegTracingSubscriber {
+    // We toggle interest on and off depending on whether anyone is listening
+    fn register_callsite(&self, metadata: &tracing::Metadata<'_>) -> Interest {
+        Interest::sometimes()
+    }
+
+    // If the mjpeg server is behind, disable subscription for a bit
+    fn enabled(&self, metadata: &tracing::Metadata<'_>) -> bool {
+        match self.visitor.try_lock() {
+            Ok(visitor) => !visitor.mjpeg_server.is_full(),
```

Comment:
- [ ] what happens if there are multiple logging call-sites that are trying to display different-sized images? Is there any good way to split each of them into their own stream, and enable/disable them based on whether anyone is observing them? tracing::Metadata might have enough information for that, if we get a bit inventive.

```
pub struct Metadata<'a> {
    /// The name of the span described by this metadata.
    name: &'static str,

    /// The part of the system that the span that this metadata describes
    /// occurred in.
    target: &'a str,

    /// The level of verbosity of the described span.
    level: Level,

    /// The name of the Rust module where the span occurred, or `None` if this
    /// could not be determined.
    module_path: Option<&'a str>,

    /// The name of the source code file where the span occurred, or `None` if
    /// this could not be determined.
    file: Option<&'a str>,

    /// The line number in the source code file where the span occurred, or
    /// `None` if this could not be determined.
    line: Option<u32>,

    /// The names of the key-value fields attached to the described span or
    /// event.
    fields: field::FieldSet,

    /// The kind of the callsite.
    kind: Kind,
}
```